src/cpp/core/system/Crypto.cpp

```diff
@@ -449,59 +405,47 @@ Error generateRsaCertAndKeyPair(const std::string& in_certCommonName,
 }
 
 namespace {
-RSA* s_pRSA;
+EVP_PKEY* s_pRSA;
 std::string s_modulo;
 std::string s_exponent;
 }
 
 core::Error rsaInit()
 {
-   const int KEY_SIZE = 2048;
-   const int ENTROPY_BYTES = 4096;
-
-   const BIGNUM *bn_n;
-   const BIGNUM *bn_e;
-
-   int rnd = ::open("/dev/urandom", O_RDONLY);
-   if (rnd == -1)
+   // seed the random number generator
+   int file = ::open("/dev/urandom", O_RDONLY);
+   if (file == -1)
       return systemError(errno, "RSA /dev/urandom", ERROR_LOCATION);
 
-   char entropy[ENTROPY_BYTES];
-   if (-1 == ::read(rnd, entropy, ENTROPY_BYTES))
-   {
-      ::close(rnd);
-      return systemError(errno, "RSA read random", ERROR_LOCATION);
-   }
-   ::close(rnd);
-
-   RAND_seed(entropy, ENTROPY_BYTES);
-
-   #if OPENSSL_VERSION_NUMBER < 0x10100000L
-      s_pRSA = ::RSA_generate_key(KEY_SIZE, 0x10001, nullptr, nullptr);
-      if (!s_pRSA)
-         return getLastCryptoError(ERROR_LOCATION);
-
-      bn_n = s_pRSA->n;
-      bn_e = s_pRSA->e;
-   #else
-      BIGNUM *bn = BN_new();
-      BN_set_word(bn, RSA_F4);
- 
-      s_pRSA = RSA_new();
-      int rc = ::RSA_generate_key_ex(s_pRSA, KEY_SIZE, bn, nullptr);
-      BN_clear_free(bn);
-      if (rc != 1) {
-        RSA_free(s_pRSA);
-        return getLastCryptoError(ERROR_LOCATION);
-      }
+   char entropy[kRsaEntropyBytes];
+   int status = ::read(file, entropy, kRsaEntropyBytes);
+   int savedErrno = errno;
+   ::close(file);
    
-      RSA_get0_key(s_pRSA, &bn_n, &bn_e, nullptr);
-   #endif
+   if (status == -1)
+      return systemError(savedErrno, "RSA read random", ERROR_LOCATION);
 
-   char* n = BN_bn2hex(bn_n);
+   RAND_seed(entropy, kRsaEntropyBytes);
```

Comment:
I wonder if we should just use `RAND_poll()` here. https://www.openssl.org/docs/manmaster/man3/RAND_poll.html